lib/DDG/Meta/ZeroClickInfoSpice.pm

```diff
@@ -155,30 +151,43 @@ sub apply_keywords {
 	$stash->add_symbol('&rewrite',sub {
 		unless (defined $rewrite) {
 			if ($target->has_rewrite) {
-				$rewrite = DDG::Rewrite->new(
-					to => $zcispice_params{'to'},
-					defined $zcispice_params{'from'} ? ( from => $zcispice_params{'from'}) : (),
-					defined $zcispice_params{'proxy_cache_valid'} ? ( proxy_cache_valid => $zcispice_params{'proxy_cache_valid'} ) : (),
-					defined $zcispice_params{'proxy_ssl_session_reuse'} ? ( proxy_ssl_session_reuse => $zcispice_params{'proxy_ssl_session_reuse'} ) : (),
-					callback => $callback,
-					path => $path,
-					wrap_jsonp_callback => $zcispice_params{'wrap_jsonp_callback'},
-					wrap_string_callback => $zcispice_params{'wrap_string_callback'},
-					accept_header => $zcispice_params{'accept_header'},
-                                        error_fallback => $zcispice_params{'error_fallback'},
-				);
-			} else {
-				$rewrite = "";
+				$rewrite = create_rewrite($callback, $path, \%zcispice_params);
+			} 
+			else {
+				$rewrite = '';
 			}
 		}
 		return $rewrite;
 	});
 
+	# make these accessbile, e.g. for duckpan
+	$stash->add_symbol('&alt_rewrites', sub {
+		my %rewrites;
+		# check if we have alternate end points to add
+		if(my $alt_to = $zcispice_params{alt_to}){
+			my ($base_target) = $target =~ /^(.+::)\w+$/;
```

Comment:
We may want to ensure that a `to` is actually defined? It would be useless without, but it throws an error if that's the case:

```
Use of uninitialized value $to in substitution (s///) at /Users/zaahir/perl5/perlbrew/perls/perl-5.16.3/lib/site_perl/5.16.3/DDG/Rewrite.pm line 12.
Use of uninitialized value $to in substitution (s///) at /Users/zaahir/perl5/perlbrew/perls/perl-5.16.3/lib/site_perl/5.16.3/DDG/Rewrite.pm line 14.
Use of uninitialized value $to in pattern match (m//) at /Users/zaahir/perl5/perlbrew/perls/perl-5.16.3/lib/site_perl/5.16.3/DDG/Rewrite.pm line 16.
```